std/functional.d

```diff
@@ -716,15 +716,9 @@ Returns:
  */
 template partial(alias fun, alias arg)
 {
-    static if (is(typeof(fun) == delegate) || is(typeof(fun) == function))
-    {
-        import std.traits : ReturnType;
-        ReturnType!fun partial(Parameters!fun[1..$] args2)
-        {
-            return fun(arg, args2);
-        }
-    }
-    else
+    import std.traits : isCallable;
+    // Check whether fun is a user defined type which implements opCall or a template (void).
+    static if (is(typeof(fun) == struct) || is(typeof(fun) == class) || is(typeof(fun) == void))
```

Comment:
Instead of doing the check for void hack, there is now an [isTemplate](https://dlang.org/spec/traits.html#isTemplate) trait to check directly.